### PR TITLE
fix support for powershell and fixed some warnings when sending packets on XP

### PR DESCRIPTION
**Pull request by n1nj4sec**
tested on windows 7 & XP
The part I'm not sure of is line 77. On windows XP, my device id in the registry seems to be the DeviceID from wmi+1

### CODE REVIEW

File: scapy/arch/windows/__init__.py [hunk 1]
```diff
@@ -72,7 +74,7 @@ def _vbs_exec_code(code):
 
 def _vbs_get_iface_guid(devid):
     try:
-        # devid = str(int(devid) + 1)
+        devid = str(int(devid) + 1) # devid in the registry seems not aligned on WinXPSP3 with wmi's fields DeviceID and/or Index #TODO check if it is always the case and what happens when there are multiple interfaces on the machine
```
Comment (p-l-) [inline scapy/arch/windows/__init__.py]: I'm not sure about this one. I had a doubt (hence the commented code), but WinPcap (from wireshark) shows the available interfaces as "interface name [Microsoft's Packet Scheduler]", and the interface with index `devid` is named something like "Microsoft's Packet Scheduler", and the interface with index `devid + 1` is named "interface name".

So maybe the real interface used is the "Microsoft's Packet Scheduler" related to "interface name"?

File: scapy/arch/windows/__init__.py [hunk 2]
```diff
@@ -539,7 +540,7 @@ def sndrcv(pks, pkt, timeout = 2, inter = 0, verbose=None, chainCC=0, retry=0, m
                                     if r.answers(sentpkt):
                                         ans.append((sentpkt, r))
                                         if verbose > 1:
-                                            os.write(1, "*")
+                                            sys.stdout.write("*")
```
Comment (p-l-) [inline scapy/arch/windows/__init__.py]: This needs to be unbuffered, hence the `os.write(1, [...])` instead of `sys.stdout.write([...])`.

File: scapy/arch/windows/__init__.py [hunk 3]
```diff
@@ -663,7 +664,7 @@ def get_working_if():
     try:
         # return the interface associated with the route with smallest
         # mask (route by default if it exists)
-        return min(read_routes(), key=lambda x: x[1])[2]
+        return min(read_routes(), key=lambda x: x[1])[3]
```
Comment (p-l-) [inline scapy/arch/windows/__init__.py:667]: Totally.

File: scapy/arch/windows/__init__.py (hunk 2 above)
Comment (n1nj4sec) [inline scapy/arch/windows/__init__.py]: ok put it back or add a flush, I changed this because it won't be "printed" on code that wraps sys.stdout and not os.write

Comment (p-l-) [inline scapy/arch/windows/__init__.py]: Unless I misunderstand what you mean, a good way to "wrap" `sys.stdout` would probably be to `os.close(1)` and `os.dup2(x, 1)`: that would work regardless of the method used to write to stdout (`sys.stdout.write()`, `os.write(1, [...])` or `print`).

Comment (n1nj4sec) [inline scapy/arch/windows/__init__.py]: i meant something like : 

``` python
class NewStdout(object):
    def write(self, data):
        #anything like :
        data=modify_data(data)
        data=cipher_data(data)
        self.sock.sendall(data)
    def flush(self):
        pass
sys.stdout=NewStdout()
print "foo"
```

Comment (p-l-) [inline scapy/arch/windows/__init__.py]: Understood, but that would be another PR then.

Maybe an option for that would be to create a `conf.output` value and use it everywhere? That would be good to have @guedou's opinion here.

File: scapy/arch/windows/__init__.py (hunk 1 above)
Comment (n1nj4sec) [inline scapy/arch/windows/__init__.py]: I installed [npcap](https://github.com/nmap/npcap) and I don't have a "Packet scheduler interface". My devid in the registry is 2 and I don't even have the interface ID 2 in wmi output, only 1, 3, 4, 5, ...
